### PR TITLE
Ethereum blockchain extractor and builder

### PR DESCRIPTION
**Pull request by prabushitha**
Leveldb blockchain extractor to extract block related data from the blockchain. Builder enables to store blockchain data extracted using the extractor in an analyzable structure. 

image

### CODE REVIEW

**Comment by tharidu** [comment]
It's better to remove the code from header files

File: tools/EtherBuilder/main.cpp [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    std::cout << "-----------PARSING STARTED-----------" << std::endl;
+    std::cout << "-------------------------------------" << std::endl;
+
+    // Start timing
+    TimeCalculator timer{};
+    timer.setStart();
+
+    // Build started
+    size_t parse_start_block = info.nextBlockId;
+    size_t parse_end_block = info.nextBlockId+100000;
```
Comment (tharidu) [inline tools/EtherBuilder/main.cpp]: Remove hard coded values

**Comment by prabushitha** [comment]
@tharidu done, isolated implementation from headers

File: tools/EtherBuilder/performance_checker_sqlite.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+void transactionsWithOptionsInsertions(sqlite3 *db) {
+    createResearchdb(db);
+
+    long int startTime = getTime();
+
+    char *zErrMsg = 0;
+    int rc;
+
+//    sqlite3_exec(db, "PRAGMA synchronous=OFF", NULL, NULL, &zErrMsg);
```
Comment (tharidu) [inline tools/EtherBuilder/performance_checker_sqlite.cpp]: Remove code comments if not needed

**Comment by tharidu** [comment]
@prabushitha Can you format the cpp files using a linter?

**Comment by prabushitha** [comment]
@tharidu sure! I'll do.

**Comment by prabushitha** [comment]
@tharidu done. I've used cpplint which adhere with google C++ coding style.

**Review by tharidu — APPROVED** [review]

**Review by tharidu — APPROVED** [review]

File: tools/EtherBuilder/performance_checker_sqlite.cpp (hunk 2 above)
Comment (prabushitha) [reply tools/EtherBuilder/performance_checker_sqlite.cpp]: fixed

**Pull request merged**